docs/source/basis/typing.rst

```diff
@@ -26,21 +33,67 @@ The default attribute values for each instance are set to appropriate values.
     my_atom = MyAtom()
     assert my_atom.n == 0
 
-``typing.Optional`` attributes have a default value of ``None`` if no default is specified.
+``typing.Optional`` attributes have a default value of ``None`` if no default
+is specified.
 
 .. code-block::
 
     assert my_atom.num is None
 
-Mutable default values for ``list`` and ``dict`` are OK.
+Mutable default values for ``list`` and ``dict`` are OK as the default value will
+be copied for each new instance.
 
 .. code-block::
 
     assert my_atom.lst == [1, 2, 3]
 
-The following statements will fail static type checking and cause Atom to raise a runtime ``TypeError`` exception.
+The following statements will fail static type checking and cause Atom to raise
+a runtime ``TypeError`` exception.
 
 .. code-block::
 
     my_atom.n = "Not an integer"
     my_atom.s = 5
+
+.. note::
+
+    The above class definition is basically translated by atom into:
+
+    .. code-block::
+
+        class MyAtom(Atom):
+            s = Str("Hello")
+            lst = List(Int(), default=[1, 2, 3])
+            num = Instance(float)
+            n = Int()
+
+    One can note that when inferring members from annotations, |Instance| will
+    always be preferred over |Typed| since the object to check may define a
+    custom instance check.
+
+.. note::
+
+    By default, atom will generate runtime checks for the content of list, dict
+    and set but it will not check the content of inner containers. For example
+    for the annotation ``list[list[int]]``, atom will check that the provided
+    list contains list but it will not check that those list contains int.
+
+    The depth at which containers are validated is controlled by the metaclass
+    keyword argument `type_containers` which default to 1. To fully omit
+    validating the content of containers one can write:
+
+    .. code-block::
+
+        class MyAtom(Atom, type_containers=0):
+            lst: list[int] = [1, 2, 3]
+
+    Which will be equivalent at runtime to, but allow type checker to validate the
+    content of the list:
+
+    .. code-block::
+
+        class MyAtom(Atom):
+            s = Str("Hello")
+            lst = List(default=[1, 2, 3])
```

Comment:
This is simply to explain how atom will convert the annotations. The member representation remains in my mind the source of truth about the behavior and I find it useful to explain to people how their annotations will be interpreted so that they can look up the appropriate member if they need to clarify something.